temporalio/lib/temporalio/cancellation.rb

```diff
@@ -0,0 +1,150 @@
+# frozen_string_literal: true
+
+require 'temporalio/error'
+
+module Temporalio
+  # Cancellation representation, often known as a "cancellation token". This is used by clients, activities, and
+  # workflows to represent cancellation in a thread/fiber-safe way.
+  class Cancellation
+    # Create a new cancellation.
+    #
+    # This is usually created and destructured into a tuple with the second value being the proc to invoke to cancel.
+    # For example: `cancel, cancel_proc = Temporalio::Cancellation.new`. This is done via {to_ary} which returns a proc
+    # to issue the cancellation in the second value of the array.
+    #
+    # @param parents [Array<Cancellation>] Parent cancellations to link this one to. This cancellation will be canceled
+    #   when any parents are canceled.
+    def initialize(*parents)
```

Comment:
Out of curiosity, why did you made it plural?

Reply:
No particular reason. In Go [it's singular](https://pkg.go.dev/context#WithCancel) but in .NET [it's plural](https://learn.microsoft.com/en-us/dotnet/api/system.threading.cancellationtokensource.createlinkedtokensource?view=net-8.0#system-threading-cancellationtokensource-createlinkedtokensource(system-threading-cancellationtoken())). I could also change the name of this var to "linked" though the idea of "parent" is clearer I think.

Reply:
I see.

There's technically a risk of creating memory leaks if the listener graph is allowed to grow in a way that a high number of cancellation scopes tokens (presumably short lives) registered on a single long-lived cancellation scope. There's definitely no such case at the moment (and might probably never be any case of this worth being concerned about), but the possibility of having "multiple parents" was kind of pointing toward use cases that could have been problematic.